set_bucket_tags/app.py

```diff
@@ -0,0 +1,135 @@
+import json
+import logging
+import traceback
+
+import boto3
+import requests
+
+from crhelper import CfnResource
+
+
+MISSING_BUCKET_NAME_ERROR_MESSAGE = 'BucketName parameter is required'
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+helper = CfnResource(
+  json_logging=False, log_level='DEBUG', boto_level='CRITICAL')
+
+
+def get_s3_client():
+  return boto3.client('s3')
+
+
+def get_bucket_name(event):
+  '''Get the bucket name from event params sent to lambda'''
+  resource_properties = event.get('ResourceProperties')
+  bucket_name = resource_properties.get('BucketName')
+  if not bucket_name:
+    raise ValueError(MISSING_BUCKET_NAME_ERROR_MESSAGE)
+  return bucket_name
+
+
+def get_bucket_tags(bucket_name):
+  '''Look up the bucket tags'''
+  client = get_s3_client()
+  response = client.get_bucket_tagging(Bucket=bucket_name)
+  log.debug(f'S3 bucket tags response: {response}')
+  tags = response.get('TagSet')
+  if not tags or len(tags) == 0:
+    raise Exception(f'No tags returned, received: {response}')
+  return tags
+
+
+def get_principal_arn(tags):
+  '''Find the value of the principal arn among the bucket tags'''
+  principal_arn_tag = 'aws:servicecatalog:provisioningPrincipalArn'
+  for tag in tags:
+    if tag.get('Key') == principal_arn_tag:
+      principal_arn_value = tag.get('Value')
+      return principal_arn_value
+  else:
+    raise ValueError('Could not derive a provisioningPrincipalArn from tags')
+
+
+def get_synapse_user_name(principal_arn):
+  '''Use the synapse id embedded in the principal arn to look up the synapse user name'''
+  synapse_id = principal_arn.split('/')[-1]
+  if not synapse_id.isdigit():
+    error_msg = (f'The synapse_id {synapse_id} derived from the principal_arn'
+      f'{principal_arn} is in an unexpected format')
+    raise ValueError(error_msg)
+  synapse_url = f'https://repo-prod.prod.sagebase.org/repo/v1/userProfile/{synapse_id}'
```

Comment:
would it make sense to parameterize this?  i'm oik if it's a follow on PR though

Reply:
Do you mean to make the base url "https://repo-prod.prod.sagebase.org/repo/v1/userProfile/" configurable? Yes, good idea.